js/services/PluginRegistry.js

```diff
@@ -0,0 +1,26 @@
+define(function (require, exports) {
+
+	const REGISTER_PLUGIN_EVENT = 'registerAtlasPlugin';
+
+	const registry = new Map();
+
+	function add(type, { title, priority, html }) {
+		let registeredPluginsOfGivenType = registry.get(type) || [];
+		registeredPluginsOfGivenType.push({ title, priority: priority || 999, html });
+		registry.set(type, registeredPluginsOfGivenType);
+	}
+
+	function findByType(type) {
+		return [ ...registry.get(type) || [] ].sort((a, b) => (a.priority > b.priority) ? 1 : -1);
+	}
+
+	document.addEventListener(REGISTER_PLUGIN_EVENT, e => {
```

Comment:
I'm interested in seeing what your plan is for using the event notification to lead to a plugin registration.  Initially, I thought this event was used to notify when a plugin has been added to atlas (so you can handle new plugins at run-time), but now i see it's responding to the event to add a plugin.  This was unexpected to me, and was wondering what the long-term goal of this function is.

Reply:
The goal is to be able to create a plugin which doesn't need Atlas as an explicit dependency (doesn't reference the `PluginRegistry` in `define`; what is more, the `PluginRegistry` could be loaded only via `optional` loader, otherwise bundling will fail due to lack of actual file in plugin's project) and decouple things. E.g. of usage:
```
document.dispatchEvent(new CustomEvent('registerAtlasPlugin', { detail: {
  type: ...,
  plugin: {
    title: ...,
    html: ...
  }
}}));
```